### PR TITLE
[tests][e2e] Add a test for `Widget#scroll_to_widget()`

### LINKED ISSUE #489: [tests] See if we can test the "scroll to widget" behaviour in a e2e test

### PR DESCRIPTION
**Pull request by DrBenton**
In a nutshell the test does the following:
- Creating 10 Placeholders in a vertical container, in a viewport which has a size that can only display 5 of them
- Scrolling to one in particular, with and without animation

closes #489 

### TIL: the transition timing is not so deterministic ^^

In one of the test cases I wanted to check that there is well and truly a transition, by checking what the screen is looking like after approximately half the duration of the transition.
My reasoning what that after having waited for approximately half of the scrolling duration, we should see the middle Placeholders as we're scrolling towards the last of them.
However, it seems that the state of the screen at this "halfway there" timing is not really deterministic, as some placeholders in the middle were visible or not depending on the environment.

Here for example the environments all saw the same placeholders at half the transition time - excepted Python 3.7 on macOS :shrug: 
Screenshot from 2022-05-11 09-38-31

As a result I had to make this test case's assertions a bit looser - which seem to do the job! :slightly_smiling_face:

### CODE REVIEW

File: tests/utilities/test_app.py [hunk 1]
```diff
@@ -49,16 +58,18 @@ def compose(self) -> ComposeResult:
     def in_running_state(
         self,
         *,
-        initialisation_timeout: float = 0.1,
-    ) -> AsyncContextManager[Capture]:
+        waiting_duration_after_initialisation: float = 0.001,
```
Comment (DrBenton) [inline tests/utilities/test_app.py:61]: Good news! :star2: 
I was (arbitrarily) waiting for 100ms before yielding the app in a "running state", but it turns out that 1ms is enough :rocket:  - which will make most of our integration tests much faster to run! :slightly_smiling_face: 

(expected those that use animation, like some of the test cases of this new test I'm adding in this PR)

File: pyproject.toml [hunk 2]
```diff
@@ -47,6 +47,9 @@ includes = "src"
 [tool.pytest.ini_options]
 asyncio_mode = "auto"
 testpaths = ["tests"]
+markers = [
```
Comment (DrBenton) [inline pyproject.toml:50]: cleaner to see our custom markers properly registered amongst other ones: :slightly_smiling_face: 
Screenshot from 2022-05-11 10-12-01

File: tests/utilities/test_app.py [hunk 3]
```diff
@@ -110,10 +134,18 @@ def __init__(self, *, width: int, height: int):
             file=file,
             width=width,
             height=height,
-            force_terminal=True,
+            force_terminal=False,
             legacy_windows=False,
         )
 
+    @property
+    def file(self) -> io.StringIO:
```
Comment (DrBenton) [inline tests/utilities/test_app.py:142]: just to make autocompletion better when working with these test classes :-)

File: tests/utilities/test_app.py [hunk 4]
```diff
@@ -110,10 +134,18 @@ def __init__(self, *, width: int, height: int):
...
+        return cast(io.StringIO, self._file)
+
+    @property
+    def is_dumb_terminal(self) -> bool:
```
Comment (DrBenton) [inline tests/utilities/test_app.py:146]: let's force this test Console to always be seen as "not dumb"

File: sandbox/vertical_container.py [hunk 5]
```diff
@@ -79,10 +79,10 @@ async def action_add_placeholder(self):
         placeholders = self.query("Placeholder")
         placeholders_count = len(placeholders)
         placeholder = Placeholder(
-            id=f"placeholder_{placeholders_count+1}",
-            name=f"Placeholder #{placeholders_count+1}",
+            id=f"placeholder_{placeholders_count}",
+            name=f"Placeholder #{placeholders_count}",
         )
-        root = self.query_one("#root")
+        root = self.get_child("root")
```
[the remark this answers is not in the document]
Comment (darrenburns) [reply sandbox/vertical_container.py:85]: I think the thinking behind this name as far as I can recall was that the signature is something like `get_child(id: str)`, so the `by_id` would be redundant.

[the remark this answers is not in the document]
Comment (willmcgugan) [reply sandbox/vertical_container.py:85]: Perhaps at some point we can make it keyword arg only, to enforce the `id="foo"`.

**Review by willmcgugan — APPROVED** [review]

**Comment by willmcgugan** [comment]
Great. Will add a ticket to make testing anything with time more deterministic.

**Pull request merged**